llvm/lib/CodeGen/LiveIntervals.cpp

```diff
@@ -780,7 +783,7 @@ void LiveIntervals::addKillFlags(const VirtRegMap *VRM) {
         // are actually never written by %2. After assignment the <kill>
         // flag at the read instruction is invalid.
         LaneBitmask DefinedLanesMask;
-        if (LI.hasSubRanges()) {
+        if (LI.hasSubRanges() && TRI->getNumAllocatableSubRegs(PhysReg) > 1) {
```

Comment:
I don't understand why this doesn't naturally fall out of the lanemasks for the register (e.g. above replace the TRI->regunits with MCRegUnitMaskIterator)

Reply:
Thanks for the suggestion @arsenm, this approach makes more sense!